### PR TITLE
Spec-compliant async compute for texture streaming

### PR DESCRIPTION
**Pull request by kd-11**
Reimplements the async task scheduler to work without violating spec so much with the old hacky synchronization method. This allows GeForce drivers newer than 470 to use this feature for users with large GPUs.

Fixes https://github.com/RPCS3/rpcs3/issues/9968

### CODE REVIEW

**Comment by Asinin3** [comment]
+3 AVG fps in GoW3.. but more importantly the framerate is much more stable, and the 1% lows are far better.
rpcs3async_KbGR6PlGdK
rpcs3_tSapWSPT1R
However, the top-half of the screen is corrupted during pre-rendered cutscenes
rpcs3async_CuH7YHAqrH
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/8053155/RPCS3.log.gz)

**Comment by kd-11** [comment]
> Uncharted 1 is still broken

Not trying to fix any games, just that it doesn't cause GPU hangs and works well with compatible titles.

**Comment by kd-11** [comment]
I have an idea what to do about the corruption, let me push something experimental for people to test.
We can just have the host GPU tell CELL when textures are ok to swap out.

**Comment by kd-11** [comment]
In the end, I did not need the host-sync method. It adds some overhead that eats into the performance gains.
@Asinin3 @hvhbot Check if the corrupted games are fixed now.

**Comment by Yahfz** [comment]
image
Corruption is gone

**Comment by Jonathan44062** [comment]
image
Now the map in Ratchet and Clank Future: ToD renders correctly with Async texture streaming enable

File: rpcs3/Emu/system_config.h [hunk 1]
```diff
@@ -168,7 +168,7 @@ struct cfg_root : cfg::node
 			cfg::_bool asynchronous_texture_streaming{ this, "Asynchronous Texture Streaming 2", false };
 			cfg::_bool fsr_upscaling{ this, "Enable FidelityFX Super Resolution Upscaling", false, true };
 			cfg::uint<0, 100> rcas_sharpening_intensity{ this, "FidelityFX CAS Sharpening Intensity", 50, true };
-			cfg::_enum<vk_gpu_scheduler_mode> asynchronous_scheduler{ this, "Asynchronous Queue Scheduler", vk_gpu_scheduler_mode::device };
+			cfg::_enum<vk_gpu_scheduler_mode> asynchronous_scheduler{ this, "Asynchronous Queue Scheduler", vk_gpu_scheduler_mode::safe };
```
Comment (Megamouse) [inline rpcs3/Emu/system_config.h:171]: missing case VulkanAsyncSchedulerDriver for dropdown translations in emu_settings::GetLocalizedSetting

**Pull request merged**